pkg/sensors/exec/exec_test.go

```diff
@@ -945,23 +948,125 @@ func TestExecProcessCredentialsSuid(t *testing.T) {
 	observertesthelper.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
 	readyWG.Wait()
 
+	noCredsChange := ec.NewProcessCredentialsChecker().
+		WithUid(0).WithEuid(0).WithSuid(0).WithFsuid(0).
+		WithGid(0).WithEgid(0).WithSgid(0).WithFsgid((0))
+	procExecNoPrivsChanged := ec.NewProcessChecker().
+		WithBinary(sm.Full(testBin)).WithProcessCredentials(noCredsChange).WithBinaryProperties(nil)
+	execNoPrivsChangedChecker := ec.NewProcessExecChecker("exec").WithProcess(procExecNoPrivsChanged)
 	if err := exec.Command(testBin).Run(); err != nil {
 		t.Fatalf("Failed to execute '%s' binary: %s\n", testBin, err)
 	}
+	/* Executing a setuid and setgid to root but we are already running as root
+	 * so no privilege change should be detected, same filters as above apply.
+	 */
+	if err := os.Chown(testSuid, 0, 0); err != nil {
+		t.Fatalf("Chown() on '%s' binary error: %s\n", testSuid, err)
+	}
+	if err := os.Chmod(testSuid, 0754|os.ModeSetuid|os.ModeSetgid); err != nil {
+		t.Fatalf("Chown() on '%s' binary error: %s\n", testSuid, err)
+	}
+	procExecSetuidRootNoPrivsChanged := ec.NewProcessChecker().
+		WithBinary(sm.Full(testSuid)).WithProcessCredentials(noCredsChange).WithBinaryProperties(nil)
+	execSetuidRootNoPrivsChangedChecker := ec.NewProcessExecChecker("exec").WithProcess(procExecSetuidRootNoPrivsChanged)
+	if err := exec.Command(testSuid).Run(); err != nil {
+		t.Fatalf("Failed to execute '%s' binary: %s\n", testSuid, err)
+	}
+
+	if err = syscall.Setuid(0); err != nil {
+		t.Fatalf("Failed to restore uid to 0 :  %s\n", err)
+	}
+	if err = syscall.Setgid(0); err != nil {
+		t.Fatalf("Failed to restore gid to 0 :  %s\n", err)
+	}
```

Comment:
What is this restoring?
I'm not seeing any Setuid() calls or Setgid calls in this test. There are some in the test below, so maybe that's what it tries to fix? In any case, I think we should add a t.Cleanup() for resetting the set{u,g}ids wherever we change them so that t is clear (i) why the happen, (ii) that the set values will be restored.

Reply:
Alright moved the setuid and setgid restore logic into t.Cleanup() for the tests that need it, for rest remove it as it useless, thank you !